### DISCUSSION TITLE
Zod schema.parse method return Partial object (every field is nullable) in nestjs app

### DISCUSSION THREAD

**Discussion by topperspal** [colinhacks/zod#2771] [Q&A] [answered]
Suppose a user schema as follows
```js
import { z } from "zod";

const UserSchema = z.object({
  name: z.string(),
});

type User = z.infer<typeof UserSchema>;
```

Here type **User** is equivalent to

```js
type User = {
    name?: string;
}
```

instead of 
```js
type User = {
    name: string;
}
```

What is wrong with zod here, I want name field to be non nullable but it is not here? It works pretty much fine on both simple **nodejs app** and **deno**, but it does not work in **nestjs** app.

## Steps to reproduce

1. Create a new nest app
```terminal
nest nest testzod
```

2. create a schema file with user schema
```js 
import { z } from "zod";

const UserSchema = z.object({
  name: z.string(),
});

type User = z.infer<typeof UserSchema>;
```

3. Check it

**Comment by remiconnesson** [comment, marked as answer]
I have the same issue actually but not in a nestjs app.

EDIT: ok I found the problem, zod requires strict mode (cf: https://zod.dev/?id=requirements )
